src/main.rs

```diff
@@ -96,6 +110,55 @@ fn main() -> anyhow::Result<()> {
             println!("{}", program_hash);
             Ok(())
         }
+        SubCommand::VerifyFromRepo {
+            solana_program_path,
+            repo_url,
+            program_id,
+            connection_url,
+            base_image,
+        } => {
+            // Get source code from repo_url
+            let base_name = run_fun!(basename $repo_url)?;
+            run_fun!(git clone $repo_url /tmp/solana-verify/$base_name)?;
+            run_fun!(cd /tmp/solana-verify/$base_name)?;
+
+            // Get the absolute build path to the solana program directory to build inside docker
+            let build_path = PathBuf::from(format!("/tmp/solana-verify/{}", base_name))
+                .join(solana_program_path.clone());
+            println!("Build path: {:?}", build_path);
+
+            // Build the code using the docker container
+            build(Some(build_path.to_str().unwrap().to_string()), base_image)?;
+
+            // Get the hash of the build
+            let executable_path =
+                run_fun!(find $solana_program_path/target/deploy -type f -name "*.so")?;
```

Comment:
I think this could potentially result in multiple outputs

Reply:
Good call. I added an optional `name_of_program` flag for repos that contain multiple programs. 

In the latest commit, also added:
-  add flag for build-bpf instead of build-sbf
- refactor into functions for a bit cleaner code